internal/relationtuple/transact_server.go

```diff
@@ -221,6 +226,11 @@ func (h *handler) patchRelationTuples(w http.ResponseWriter, r *http.Request, _
 			h.d.Writer().WriteError(w, r, herodot.ErrBadRequest.WithError("relation_tuple is missing"))
 			return
 		}
+		if err := d.RelationTuple.Validate(); err != nil {
+			h.d.Writer().WriteError(w, r, errors.WithStack(herodot.ErrBadRequest.WithError(err.Error())))
```

Comment:
```suggestion
			h.d.Writer().WriteError(w, r, err)
```

The error returned by validate should already be a `herodot.ErrBadRequest` with the right details and stack.